src/lib.rs

```diff
@@ -184,32 +188,35 @@ pub trait MysqlShim<W: Write> {
     fn on_init(&mut self, _: &str, _: InitWriter<'_, W>) -> Result<(), Self::Error> {
         Ok(())
     }
+
+    /// Provides the TLS configuration, if we want to support TLS.
+    fn tls_config(&self) -> Option<&TlsConfig> {
+        None
```

Comment:
Is there a sensible default configuration we can supply here?

Reply:
I am not sure there is. Also, `None` really means "don't support TLS" which is a reasonable default in itself I think.

Reply:
I guess that's the part I'm pushing back on — as a user, I would expect TLS to work "out of the box", as it does with basically all clients these days, using the system credentials. Presumably `rustls` has a default TLS config too?

Reply:
Oh, wait, I'm being dumb. This is a server, not a client, so configuration is definitely non-trivial, needing a server certificate and such. Ignore me :sweat_smile: